### ISSUE TITLE
Task.json is not identified if root directory is opened instead of project directory. 

### ISSUE THREAD

**Issue by foo-baar** [Microsoft/vscode#22990]
I am experiencing this rather unusual behaviour:

I have simple tasks in my task file currently located at:
![screen shot 2017-03-22 at 9 44 51 am](https://cloud.githubusercontent.com/assets/13655874/24181937/3e5298fa-0ee4-11e7-9518-52ee756506dd.png)

tasks:
```
{
            "taskName": "ChromeRun",
            "isBuildCommand": true,
            "showOutput":"always",
            "command": "Chrome",
            "osx": {
                "command": "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"
            },
            "args": [
                "${file}"
            ]
        }
```
Now things work well if I Open project at level of geoLocation directory inside vscode, but when I open the folder at root level (as shown in screen shot above) and try to execute tasks, the compiler fails to recognise my task.json file and asks me to create on in root directory (.vscode directory).

Considering the mapping model this is understandable, But what if I want to maintain tasks at each project level, Since all project has different configurations & I don't want all of them to be executed.

(Lack of documentation at https://code.visualstudio.com/docs/editor/tasks routed me this forum, please point to me the more relevant one, in case I am lost).

Thanks much.

**Comment by dbaeumer** [comment]
This is how workspace settings are defined right now. The have to live in the root of the folder opened (the task framework doesn't look for a tasks.json file). It simple reads the `tasks` settings from VS Code's settings service.

Changing this will require quite some work in the settings story.

**Comment by foo-baar** [comment]
hmm, We'll need this down the road, for scenario like: I am working inside a project and want to run & transpile .sass of only that project before build\run, current setup will force me to do it for all the projects under that root.

Somewhat understood as we support only limited scope as of now.
Thanks @dbaeume 👍🏻

**Comment by dbaeumer** [comment]
I am closing this. The tasks.json depends on our setting story which task.js should follow.

Scenarios like the one are now better supported using multi root workspaces which is available 1.18 released yesterday
